src/hooks/useErcBalance.ts

```diff
@@ -0,0 +1,47 @@
+import { queryOptions, useQuery } from '@tanstack/react-query'
+import { type Address, type Client, getContract } from 'viem'
+
+import { createQueryKey } from '~/react-query'
+import { erc20ABI } from '~/utils/abi'
+
+import { useClient } from './useClient'
+
+type UseErcBalanceParameters = {
+  address: Address
+  erc: Address
+}
+
+export const getErcBalanceQueryKey = createQueryKey<
+  'erc-balance',
+  [key: Client['key'], args: UseErcBalanceParameters]
+>('erc-balance')
+
+export function useErcBalanceQueryOptions({
+  erc,
+  address,
+}: UseErcBalanceParameters) {
+  const client = useClient()
+  return queryOptions({
+    enabled: Boolean(address),
+    queryKey: getErcBalanceQueryKey([client.key, { erc, address }]),
+    async queryFn() {
+      const contract = getContract({
+        address: erc,
+        abi: erc20ABI,
+        publicClient: client,
+      })
+
+      return Promise.all([
+        contract.read.name(),
+        contract.read.symbol(),
+        contract.read.decimals(),
+        contract.read.balanceOf([address]),
```

Comment:
maybe for next pr: it's probably better to just leave this as only the balanceOf call, and then add the name/symbol/decimals metadata to the tokens store? 

It should attempt to validate that the address is an erc20 (not sure how really), and maybe include if it's 721, which can save the tokenUri if it exists? q is whether we want multiple stores for 20/721/1155 or if it's possible to keep in 1 and just filter it.